src/Client.re

```diff
@@ -397,3 +397,30 @@ let mutation =
   |> Wonka.take(1)
   |> Wonka.toPromise;
 };
+
+let readQuery =
+    (
+      ~client,
+      ~request,
+      ~fetchOptions=?,
+      ~requestPolicy=?,
+      ~url=?,
+      ~pollInterval=?,
+      ~meta=?,
```

Comment:
One thing to note – we recently added support for the full spectrum of options that can be passed in a `Partial<OperationContext>`. These new additions include `fetch` (custom `fetch` implementation to use), `additionalTypenames` (let the operation know that it depends on specific `__typenames`, used mostly by the standard document cache), `suspense` (a boolean flag to enable suspense), and `preferGetMethod` (instruct GraphQL operations to be executed with `GET` rather than `POST` as the http method).

You can look at `executeQuery` and match that signature (since it's the same) and we should be good to go! I'm also realizing I need to update the bindings for `query` and `mutation` while I'm here as well.

Reply:
Thanks for the detailed explanation 😊  I indeed took inspiration from the `query` one so I missed those extra options